### DISCUSSION TITLE
Fastify Plugin for spinning Node Child Process

### DISCUSSION THREAD

**Discussion by Easy-Cloud-in** [fastify/fastify#5022] [Q&A] [answered]
Can any one help to create a Fastify plugin for spinning Node Child process and hook it to each HTTPS methods.

Edit:
I am writing a multitenant app, where each tenant has their own database. So, I want to spawn a node child process for each client to access their database. whether to write a plugin or a middleware and hook it to preHandler.

**Comment by Uzlopak** [comment]
As I wrote you in discord, you need to elaborate more. With this description you get nowhere.

**Reply by Easy-Cloud-in** [reply]
I am writing a multitenant app, where each tenant has their own database.  So, I want to spawn a node child process for each client to access their database. whether to write a plugin or a middleware and hook it to preHandler.

**Reply by metcoder95** [reply]
Why a `child_process` and not just a new connection or a proxy that sits in-front of the database?

**Reply by Easy-Cloud-in** [reply]
Yes after,  doing some reach, I planned to use bgBouncer for connection pooling. I hope this will take care of horizontal scaling in AWS Fargate, where I am planning to run my app. Thank you for your time.

**Comment by Eomm** [comment, marked as answer]
I would start from here: https://github.com/piscinajs/piscina

It seems you need a `fastify-piscina` plugin

**Reply by Easy-Cloud-in** [reply]
Can you please explain what this plugin piscina does? Can you explain how it works for my use case?
